File: src/etch/bindings/csharp/transport/ListenerHandler.cs
using System;
using System.Text;
using System.Net.Sockets;
using System.Collections.Generic;

namespace Etch.Transport
{
    /// <summary>
    /// Interface used to report listener events
    /// </summary>
    public interface ListenerHandler : SourceHandler<Source>
    {
        /// <summary>
        /// Reports that the listener has accepted a connection.
        /// </summary>
        /// <param name="s">the socket of the connection.</param>
        /// Exception:
        ///     throws Exception

        void Accepted( Socket s );
    }
}

File: binding-ruby/src/main/ruby/transport/listener_handler.rb
require 'socket'
 #
 # Interface used to report listener events.
 #
module ListenerHandler include
			SourceHandler
	#
	# Reports that the listener has accepted a connection.
	# @param t event originator
	# @param s the socket of the connection.
	# @throws Exception 
	#
	def accepted( t, s ) 
		raise "Subclass responsibility"
	end
end
